src/main/kotlin/camp/nextstep/edu/racingcar/interfaces/CarTrackSnapshot.kt

```diff
@@ -0,0 +1,15 @@
+package camp.nextstep.edu.racingcar.interfaces
+
+import camp.nextstep.edu.racingcar.racing.Car
+import camp.nextstep.edu.racingcar.racing.CarRaceEvent
+
+class CarTrackSnapshot(car: Car, events: List<CarRaceEvent>) {
+
+    private val carName = car.name
+    private val track = CarRaceEvent.merge(events)
+    private val traces = track.traces().asSequence().joinToString(" ") { if (it) "⎼" else "" }
```

Comment:
```suggestion
    private val traces = track.traces()joinToString(" ") { if (it) "⎼" else "" }
```
코틀린 sequence 는 자바의 스트림과 비슷합니다. 대량의 컬렉션에서는 장점이지만 소수의 컬렉션에서는 오버헤드가 있습니다.
"kotlin list vs sequence" 에 대해 검색해 보시고 이해하면 좋을것 같습니다.

Reply:
Track::traces 리턴 타입을 `Iterator` 로 두어 큰 고민 없이 사용하게 된 것 같습니다 🥲
글을 읽고 생각해보니,, Track 이 아주 긴 경우에는 유효하겠지만, 그렇지 않은 경우에는 불필요할 수 있겠네요 🤔
Iterator 보다는 Collection 를 리턴하도록 하는 것이 좋을 것 같다고 생각하여 수정했습니다!